src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,40 +1,78 @@
-import React from 'react';
+import React, { useMemo, useState } from 'react';
 import { Loader } from '../Loader';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { filterTodo } from '../../features/filterTodo';
+import { actions } from '../../features/currentTodo';
 
 export const TodoModal: React.FC = () => {
+  const [currentUser, setCurrentUsers] = useState<User>();
+  const currentTodo = useAppSelector(state => state.currentTodo);
+  const todos = useAppSelector(filterTodo.getFilteredTodos);
+  const dispatch = useAppDispatch();
+
+  const removeCurrentTodo = () => {
+    dispatch(actions.removeTodo());
+  };
+
+  useMemo(() => {
+    if (currentTodo) {
+      getUser(currentTodo?.userId).then((user) => setCurrentUsers(user));
```

Comment:
Remove the optional chaining operator '?' from 'currentTodo?.userId' since it is already checked in the if condition.